### PR TITLE
feat: return list of dictionaries for execute streaming sql

### LINKED ISSUE #958: Is there an option to return execute_sql results as a list of dictionaries?
right now the results returned from `snapshot.execute_sql()` gets returned as a list of arrays.  Is there an existing configuration to change the return format of the StreamedResultSet object?

I would like to use keys instead of row indices when pulling values out for further processing. If i ever had to update the running query I would have to make sure my indices are correct. I can easily convert the row to dictionary but want to see if there's a configuration i'm missing.

### PR DESCRIPTION
**Pull request by asthamohta**

### CODE REVIEW

File: google/cloud/spanner_v1/streamed.py [hunk 1]
```diff
@@ -190,6 +190,19 @@ def one_or_none(self):
         except StopIteration:
             return answer
 
+    def fetch_as_dictionary_list(self):
```
Comment (surbhigarg92) [inline google/cloud/spanner_v1/streamed.py:193]: Name can be better . Probably "to_dictionary" ?

File: google/cloud/spanner_v1/streamed.py [hunk 2]
```diff
@@ -190,6 +190,19 @@ def one_or_none(self):
...
+        rows = []
```
Comment (surbhigarg92) [inline google/cloud/spanner_v1/streamed.py:194]: Add documentation

File: google/cloud/spanner_v1/streamed.py [hunk 3]
```diff
@@ -190,6 +190,19 @@ def one_or_none(self):
...
+        for row in self:
```
Comment (surbhigarg92) [inline google/cloud/spanner_v1/streamed.py:195]: What happens if rows are not fully processed when this method is called ?

File: google/cloud/spanner_v1/streamed.py (hunk 1 above)
Comment (rajatbhatta) [reply google/cloud/spanner_v1/streamed.py:193]: to_dictionary would mean it's a dict but the return type here is a list of dict.

Comment (rajatbhatta) [reply google/cloud/spanner_v1/streamed.py:193]: Simply `dict_list`? Similar to method name `one`?

Comment (rajatbhatta) [reply google/cloud/spanner_v1/streamed.py:193]: Does to_dict_list look good?

File: google/cloud/spanner_v1/streamed.py (hunk 3 above)
Comment (asthamohta) [reply google/cloud/spanner_v1/streamed.py:195]: There will be a delay in returning the result in those cases but always the full result will be returned

**Review by rajatbhatta — APPROVED** [review]

**Pull request merged**